File: Number of Good Components.cpp
class Solution {
  public:
    void dfs(int v, int &vertices, int &edges, vector<vector<int>> &adj,
             vector<int> &visited) {
        visited[v] = 1;
        vertices++;
        edges += adj[v].size();
        for (auto to : adj[v]) {
            if (!visited[to]) {
                dfs(to, vertices, edges, adj, visited);
            }
        }
    }
    int findNumberOfGoodComponent(int V, vector<vector<int>>& adj) {
        int ans = 0;
        vector<int> visited(V + 1, 0);

        for (int i = 1; i <= V; i++) {
            if (!visited[i]) {
                int vertices = 0, edges = 0;
                dfs(i, vertices, edges, adj, visited);
                edges /= 2;
                if (edges == (vertices * (vertices - 1)) / 2) ans++;
            }
        }
        return ans;
    }
};

File: Number of Good Components.py
import collections
from collections import deque
from collections import defaultdict

class Solution:
    def dfs(self,v,adj,visited):
        visited[v]=1
        self.vertices+=1
        self.edges+=len(adj[v])
        for to in adj[v]:
            if not visited[to]:
                self.dfs(to,adj,visited)
                
        
    def findNumberOfGoodComponent(self, V, adj):
        ans=0
        visited=[0]*(V+1)
        for i in range(1,V+1):
            if not visited[i]:
                self.vertices=0
                self.edges=0
                self.dfs(i,adj,visited)
                self.edges//=2
                if (self.edges==(self.vertices*(self.vertices-1))//2):
                    ans+=1
        return ans
